src/vm/dllimport.cpp

```diff
@@ -7047,7 +7055,6 @@ HINSTANCE NDirect::LoadLibraryModule(NDirectMethodDesc * pMD, LoadLibErrorTracke
     bool libNameIsRelativePath = Path::IsRelative(wszLibName);
     if (hmod == NULL)
     {
-#ifndef FEATURE_CORECLR
         // First checks if the method has DefaultDllImportSearchPathsAttribute. If method has the attribute
```

Comment:
Based upon https://msdn.microsoft.com/en-us/library/system.runtime.interopservices.dllimportsearchpath(v=vs.110).aspx, what the Interop implementation does is a policy determined by the values of this attribute. The host has simply provided the data (via NATIVED_DLL_SEARCH_PATHS) to the runtime. 

We should look at the how the table at https://msdn.microsoft.com/en-us/library/system.runtime.interopservices.dllimportsearchpath(v=vs.110).aspx would be processed for native code. It appears to me that even if native code carried a DLL applocal and this attribute was used, there are scenarios where machine wide serviced versions of the binary could get skipped in preference over applocal.

Thoughts?